app/screens/post_options/index.ts

```diff
@@ -33,27 +33,24 @@ type EnhancedProps = WithDatabaseArgs & {
     location: string;
 }
 
-const canEditPost = (isOwner: boolean, post: PostModel, postEditTimeLimit: number, isLicensed: boolean, channel: ChannelModel, user: UserModel): boolean => {
+const observeCanEditPost = (isOwner: boolean, post: PostModel, postEditTimeLimit: number, isLicensed: boolean, channel: ChannelModel, user: UserModel) => {
     if (!post || isSystemMessage(post)) {
-        return false;
+        return of$(false);
     }
 
-    let cep: boolean;
-
-    const permissions = [Permissions.EDIT_POST];
-    if (!isOwner) {
-        permissions.push(Permissions.EDIT_OTHERS_POSTS);
-    }
-
-    cep = permissions.every((permission) => hasPermissionForChannel(channel, user, permission, false));
```

Comment:
I feel this was not working at all (it was always returning true, since every element had a promise, which is distinct from null, therefore true). I tried to fix it.